Import errors with use_frameworks!

If you try to install KSCrash via CocoaPods with `use_frameworks!`, in a Podfile like this:
``` ruby
platform :ios, '8.0'
use_frameworks!

target 'PodsTest' do
  pod 'KSCrash', :git => 'git@github.com:kstenerud/KSCrash.git', :tag => '1.0.2'
end
```

You'll get an error like this one:
```
While building module 'KSCrash':
In file included from <module-includes>:1:
KSCrash/KSCrash.framework/Headers/KSCrash-umbrella.h:19:9: fatal error: 'KSCrashAdvanced.h' file not found
#import "KSCrashAdvanced.h"
        ^
1 error generated.
fatal error: could not build module 'KSCrash'
#import <KSCrash/KSCrash.h>
 ~~~~~~~^
```

This appears to have started happening with #124, and to the best of my understanding it seems to be caused because the `KSCrash/Recording/Advanced` subspec marks some headers as public, but those headers aren't part of the `source_files` of the spec. (`public_header_files` is expected to be a subset of `source_files`, which I learned from CocoaPods/CocoaPods#4936.) Including these headers as `source_files` seems to fix the problem.

After fixing that, there's also a second issue if you try to use just the `Recording` subspec:
```
While building module 'KSCrash':
In file included from <module-includes>:1:
In file included from KSCrash/KSCrash.framework/Headers/KSCrash-umbrella.h:3:
KSCrash/KSCrash.framework/Headers/KSCrash.h:31:9: fatal error: 'KSCrashReportFilter.h' file not found
#import "KSCrashReportFilter.h"
        ^
1 error generated.
fatal error: could not build module 'KSCrash'
#import <KSCrash/KSCrash.h>
 ~~~~~~~^
```

This one's a bit more straightforward, `KSCrashReportFilter.h` wasn't included as a public header for the `Recording` subspec. Adding it fixes the problem.

PR to fix both of these incoming!

--- a/KSCrash.podspec
+++ b/KSCrash.podspec
@@ -12,6 +12,7 @@ Pod::Spec.new do |s|
   s.frameworks = 'Foundation'
   s.libraries = 'c++', 'z'
   s.xcconfig = { 'GCC_ENABLE_CPP_EXCEPTIONS' => 'YES' }
+  s.default_subspecs = 'Installations'
 
   s.subspec 'no-arc' do |sp|
     sp.source_files = 'Source/KSCrash/Recording/**/KSZombie.{h,m}'
@@ -34,21 +35,28 @@ Pod::Spec.new do |s|
                                     'Source/KSCrash/Recording/Sentry/KSCrashSentry.h',
                                     'Source/KSCrash/Recording/Tools/KSArchSpecific.h',
                                     'Source/KSCrash/Recording/Tools/KSJSONCodecObjC.h',
-                                    'Source/KSCrash/Recording/Tools/NSError+SimpleConstructor.h'
+                                    'Source/KSCrash/Recording/Tools/NSError+SimpleConstructor.h',
+                                    'Source/KSCrash/Reporting/Filters/KSCrashReportFilter.h'
+  end
 
-    recording.subspec 'Advanced' do |advanced|
-      advanced.public_header_files = 'Source/KSCrash/Recording/KSCrashAdvanced.h',
-                                     'Source/KSCrash/Recording/KSCrashDoctor.h',
-                                     'Source/KSCrash/Recording/KSCrashReportFields.h',
-                                     'Source/KSCrash/Recording/KSCrashReportStore.h',
-                                     'Source/KSCrash/Recording/KSSystemInfo.h',
-                                     'Source/KSCrash/Recording/KSSystemInfoC.h'
-    end
+  # This subspec is just to optionally expose the advanced headers
+  s.subspec 'RecordingAdvanced' do |advanced|
+    advanced.dependency 'KSCrash/Recording'
+    # Just add .h files; the .m files are all compiled in the Recording spec
+    advanced.source_files = 'Source/KSCrash/Recording/KSCrashAdvanced.h',
+                            'Source/KSCrash/Recording/KSCrashDoctor.h',
+                            'Source/KSCrash/Recording/KSCrashReportFields.h',
+                            'Source/KSCrash/Recording/KSCrashReportStore.h',
+                            'Source/KSCrash/Recording/KSSystemInfo.h',
+                            'Source/KSCrash/Recording/KSSystemInfoC.h'
+  end
 
-    recording.subspec 'Tools' do |tools|
-      tools.public_header_files = 'Source/KSCrash/Recording/Tools/**/*.h'
-    end
-
+  # This subspec is just to optionally expose the tools headers
+  s.subspec 'RecordingTools' do |tools|
+    tools.dependency 'KSCrash/Recording'
+    # Just add .h files; the .m files are all compiled in the Recording spec
+    tools.source_files = 'Source/KSCrash/Recording/Tools/*.h'
+    tools.exclude_files = 'Source/KSCrash/Recording/Tools/KSZombie.h'
   end
 
   s.subspec 'Reporting' do |reporting|

